sources/packages/backend/workflow/test/2022-2023/fulltime-assessment/fulltime-assessment-2022-2023.e2e-spec.ts

```diff
@@ -0,0 +1,122 @@
+import { OfferingIntensity } from "@sims/sims-db";
+import {
+  AssessmentConsolidatedData,
+  AssessmentModel,
+  CalculatedAssessmentModel,
+} from "../../models";
+import { ZBClient } from "zeebe-node";
+import { getFakeAssessmentConsolidatedData } from "../../utils";
+import { PROGRAM_YEAR } from "../constants/program-year.constants";
+
+describe(`E2E Test Workflow  fulltime-assessment-${PROGRAM_YEAR}`, () => {
+  let zeebeClientProvider: ZBClient;
+  beforeAll(async () => {
+    zeebeClientProvider = new ZBClient();
+  });
+
+  it("Should generate expected fulltime assessment values for a single and independent student.", async () => {
+    const assessmentConsolidatedData = getFakeAssessmentConsolidatedData(
+      OfferingIntensity.fullTime,
+      PROGRAM_YEAR,
+      "2023-02-01",
+      "2023-05-24",
+    );
+    const expectedAssessmentData: AssessmentModel = {
+      weeks: assessmentConsolidatedData.offeringWeeks,
+      tuitionCost:
+        assessmentConsolidatedData.offeringActualTuitionCosts +
+        assessmentConsolidatedData.offeringMandatoryFees,
+      childcareCost: 0,
+      livingAllowance: 1672,
+      totalAssessedCost: 27672,
+      totalFamilyIncome: 40000,
+      totalFederalAward: 5600,
+      otherAllowableCost: 0,
+      transportationCost: 0,
+      secondResidenceCost: 0,
+      totalAssessmentNeed: 55049.19807692308,
+      booksAndSuppliesCost: 5000,
+      totalProvincialAward: 11037.417692307692,
+      alimonyOrChildSupport: 0,
+      federalAssessmentNeed: 26000,
+      exceptionalEducationCost:
+        assessmentConsolidatedData.offeringExceptionalExpenses,
+      provincialAssessmentNeed: 27593.544230769232,
+      parentAssessedContribution: null,
+      partnerAssessedContribution: null,
+      studentTotalFederalContribution: 216.34615384615384,
+      studentTotalProvincialContribution: 78.45576923076923,
+    } as AssessmentModel;
+
+    const calculatedAssessment =
+      await zeebeClientProvider.createProcessInstanceWithResult<
+        AssessmentConsolidatedData,
+        CalculatedAssessmentModel
+      >(`fulltime-assessment-${PROGRAM_YEAR}`, assessmentConsolidatedData);
+    // TODO: totalFederalContribution and totalProvincialContribution needs to be validated
+    // once it is fixed in bpmn.
+    expect(calculatedAssessment.variables.offeringWeeks).toBe(
```

Comment:
Are we individually testing the expected values one by one because we are not covering the entire result object?
Asking to know if there is not possibility to use the `.toBe({ var1: 1123, var2: 4444})` check.

Reply:
As discussed the result object consists of variables which are input variables + result variables.

I tried an approach to validate the partial object and found this way. This worked perfectly on happy path.

![image](https://user-images.githubusercontent.com/54600590/225357533-15726673-6b22-4214-8bce-12a13941fc8e.png)

But when I tested for failure. the test failure message wasn't clear and appropriate IMO.
![image](https://user-images.githubusercontent.com/54600590/225357936-b2f1abd9-109c-4dda-bd93-08e3fcd54fc4.png)

![image](https://user-images.githubusercontent.com/54600590/225358177-b4bd2f0d-b653-4a8a-b3d7-205bfd8ee454.png)

But with the current code the place of failure was more clear to identify(tested with hardcoded offering weeks 17).
![image](https://user-images.githubusercontent.com/54600590/225358596-59de0716-42b4-42b7-b97d-e2a3d2aa50ce.png)

So I thought we can discuss this as a team and if changes required I can make in upcoming PR.

Reply:
mm... I didn't get it. here `received` and `expected` indicate what?

Reply:
I am ok with the way that it is right now and thanks for sharing the additional information.

Reply:
Expected is the value that we expect and Received is the actual output(of workflow in this case).

Reply:
i know the meaning @dheepak-aot. i was asking about this example. is it elements of the object?